### PR TITLE
Deletes DataSource, instead binds graphs directly to GuiStateContainer.

### PR DESCRIPTION
**Pull request by jkff**
This addresses https://github.com/RespiraWorks/VentilatorSoftware/issues/6, except of course without real data - that's up to https://github.com/RespiraWorks/VentilatorSoftware/pull/226.

### CODE REVIEW

File: gui/gui_state_container.h [hunk 1]
```diff
@@ -52,6 +59,10 @@ class GuiStateContainer {
     return history_.GetHistory();
   }
 
+public slots:
```
Comment (jlebar) [inline gui/gui_state_container.h:62]: wat.

Does qt have...a custom C++ preprocessor?

File: gui/gui_state_container.cpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+                controller_status.sensor_readings.pressure_cm_h2o));
+    flow_points.append(
+        QPointF(neg_millis_ago * 0.001,
+                controller_status.sensor_readings.flow_ml_per_min));
+    tv_points.append(QPointF(neg_millis_ago * 0.001,
+                             controller_status.sensor_readings.volume_ml));
+  }
+  qobject_cast<QXYSeries *>(pressure_series)->replace(pressure_points);
+  qobject_cast<QXYSeries *>(flow_series)->replace(flow_points);
+  qobject_cast<QXYSeries *>(tv_series)->replace(tv_points);
```
Comment (jlebar) [inline gui/gui_state_container.cpp:26]: Does this copy the points vectors?  Or is there even more Qt magic here, it's doing a `swap` or something?

File: gui/main.qml [hunk 3]
```diff
@@ -66,7 +66,7 @@ Item
         ScopeView
         {
             id: flowView
-            name: "Flow [mL]"
+            name: "Flow [mL/min]"
```
Comment (jlebar) [inline gui/main.qml:69]: Not to be changed here, fwiw I think this is probably the wrong unit.  TV is on the order if 1L per breath, and you do roughly a dozen breaths/min, so L/min seems like the right unit.  @SarahAdelaide may have an opinion, having gone through this constants exercise?

File: gui/gui_state_container.cpp (hunk 2 above)
Comment (jkff) [reply gui/gui_state_container.cpp:26]: Under the hood it's an assignment operator. I *think* it should support move-assignment so I'm going to add that.

File: gui/gui_state_container.h (hunk 1 above)
Comment (jkff) [reply gui/gui_state_container.h:62]: Yup. https://en.wikipedia.org/wiki/Meta-object_System

File: gui/main.qml (hunk 3 above)
Comment (jkff) [reply gui/main.qml:69]: Ack - I'm just using the unit specified in the proto.

**Pull request merged**